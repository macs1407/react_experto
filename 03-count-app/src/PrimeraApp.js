import React from 'react';
import PropTypes from 'prop-types';
import CounterApp from './CounterApp';

// Hay dos tipos de componentes, los que estan basados por clases y por funciones
// Ahora se maneja en base de funciones, FUNCTIONAL COMPONENTS

// Las propiedades se manda de componente a componente
    // Se puede establecer un valor por defecto si no se envia esa propiedad
const PrimeraApp = ({saludoPadre, subtitulo})=>{
    const saludo = "Hola mundo";
    const numero = 123;
    // Si son boleanos no se imprimen, se utilizan para hacer condiciones para mostrar
    const arreglo = [1,2,3];
    // No se puede imprimir todo un objeto
    const objeto = {
        nombre:"Maikol",
        edad:31,
    }
    
    //const {saludoPadre="defecto"} = props;
    return (
        <>
            <h1>{saludoPadre}.</h1>
            <h1>{saludo} {numero}</h1>
            <p>{arreglo}</p>
            <p>{JSON.stringify(objeto)}</p>
            <h2>{subtitulo}</h2>
            <CounterApp numero={31}/>
        </>
    );
}

// Definir propiedades al componente, se puede decir que la propiedad saludo es obligatoria
PrimeraApp.propTypes = {
    saludoPadre: PropTypes.string.isRequired
}
// Definir valores a las propiedades por defecto
PrimeraApp.defaultProps = {
    subtitulo: "Soy un un subtitulo"
}

export default PrimeraApp;