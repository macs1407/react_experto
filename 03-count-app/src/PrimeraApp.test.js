import React from 'react';
import '@testing-library/jest-dom';
import PrimeraApp from './PrimeraApp';
import {shallow} from 'enzyme';

describe('Pruebas en <PrimeraApp />', () => {
    
    /* test('Probando un componente ', () => {
        const saludo = "hola";
        const {getByText} = render(<PrimeraApp saludoPadre={saludo} />);
        expect(getByText(saludo)).toBeInTheDocument()
    }); */
    
    test('Debe de mostrar <PrimeraApp /> correctamente ', () => {
        const saludo = 'hola';
       const wraper = shallow(<PrimeraApp saludoPadre={saludo} />);
       expect(wraper).toMatchSnapshot();
    });

    test('Debe de mostrar <PrimeraApp /> subtitulo, correctamente ', () => {
       const saludo = 'hola';
       const wraper = shallow(<PrimeraApp saludoPadre={saludo} subtitulo="subtitulo desde pruebas" />);
       const textoParrafo = wraper.find('h2').text();
       expect(textoParrafo).toBe("subtitulo desde pruebas");
    });
    
})
