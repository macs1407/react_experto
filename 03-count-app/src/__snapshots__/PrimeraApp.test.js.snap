// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <PrimeraApp /> Debe de mostrar <PrimeraApp /> correctamente  1`] = `
<Fragment>
  <h1>
    hola
    .
  </h1>
  <h1>
    Hola mundo
     
    123
  </h1>
  <p>
    1
    2
    3
  </p>
  <p>
    {"nombre":"Maikol","edad":31}
  </p>
  <h2>
    Soy un un subtitulo
  </h2>
  <CounterApp
    numero={31}
  />
</Fragment>
`;
